packages/@react-aria/label/src/useField.ts

```diff
@@ -48,10 +48,10 @@ export function useField(props: AriaFieldProps): FieldAria {
   let descriptionProps: HTMLAttributes<HTMLElement> = {};
   let errorMessageProps: HTMLAttributes<HTMLElement> = {};
   if (description) {
-    descriptionProps.id = descriptionId;
+    descriptionProps.id = descriptionId || errorMessageId;
```

Comment:
I don't think this is the right way to handle this. I think there is a different issue here, that useSlotId won't check to see if the slotted id exists after initial render https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/utils/src/useId.ts#L101
trying to think of other ways to handle this

Reply:
We may need to use something like a singleton pattern for id's, we've been talking about how our shared id map doesn't actually cause a re-render, so this might be the thing to do https://github.com/Light-Keeper/react-singleton-hook
then useSlotId should actually check the id is valid on every render, not just initial, it may have just been added or it may have just been removed

I'm also thinking useSlotId, instead of passing back just an id, maybe it passes back a pair of functions (hooks?), first one just returns the id state, second one, when mounted sets the id state and we know it's being used, so can dispense with the document.query, and when it's unmounted, cleans up the id state

sorry if those options are vague, this is not something I've had much time to think through

Reply:
@dkario @majornista y'all should coordinate on this

Reply:
@snowystinger We discussed it a bit before I opened the issue and developed this workaround.

Reply:
cool, I just saw @dkario assigned to himself but you opened a PR, thanks for clarifying